src/RecordManager/Base/Command/Records/Export.php

```diff
@@ -716,4 +795,134 @@ protected function writeRecord(string $record): void
         ++$this->currentBatchCount;
         file_put_contents($this->currentFile, $record . "\n", FILE_APPEND);
     }
+
+    /**
+     * Add an node to the XML record
+     *
+     * @param \SimpleXMLElement $xml       XML record
+     * @param array             $path      XML path
+     * @param string            $nodeValue Node value
+     *
+     * @return void
+     */
+    protected function addXmlNode(\SimpleXMLElement $xml, array $path, string $nodeValue): void
+    {
+        $currentNode = $xml;
+        foreach ($path as $pathPart) {
+            // Check for existing node:
+            if ($node = $this->findXmlNode($currentNode, $pathPart)) {
+                $currentNode = $node;
+                continue;
+            }
+
+            // Add a new node:
+            $currentNode = $currentNode->addChild(
+                $pathPart['nodeName'],
+                null,
+                $this->getXmlNamespaceFromPrefix($xml, $pathPart['prefix'])
+            );
+            foreach ($pathPart['attrs'] as $attr => $value) {
+                $currentNode->addAttribute($attr, $value);
+            }
+        }
+        // See https://github.com/phpstan/phpstan/issues/8236
+        // @phpstan-ignore-next-line
+        $currentNode[0] = $nodeValue;
+    }
+
+    /**
+     * Try to find a node in XML element
+     *
+     * @param \SimpleXMLElement $xml      XML node
+     * @param array             $pathPart Path item
+     *
+     * @return ?\SimpleXMLElement
+     */
+    protected function findXmlNode(\SimpleXMLElement $xml, array $pathPart): ?\SimpleXMLElement
+    {
+        $ns = $this->getXmlNamespaceFromPrefix($xml, $pathPart['prefix']);
+        foreach ($xml->children($ns, true)->{$pathPart['nodeName']} as $candidate) {
+            if ($this->xmlNodeAttributesMatch($candidate, $pathPart['attrs'])) {
+                return $candidate;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Check if node attributes match the given array of attributes
+     *
+     * @param \SimpleXMLElement $node       Node
+     * @param array             $attributes Attributes to check
+     *
+     * @return bool
+     */
+    protected function xmlNodeAttributesMatch(\SimpleXMLElement $node, array $attributes): bool
+    {
+        foreach ($node->attributes() as $attr => $value) {
+            if (($attributes[$attr] ?? null) !== $value) {
+                return false;
+            }
+            unset($attributes[$attr]);
+        }
+        return !$attributes;
+    }
+
+    /**
+     * Get namespace from prefix
+     *
+     * @param \SimpleXMLElement $xml    Node
+     * @param ?string           $prefix Prefix
+     *
+     * @return string
+     */
+    protected function getXmlNamespaceFromPrefix(\SimpleXMLElement $xml, ?string $prefix): ?string
+    {
+        $ns = null;
```

Comment:
Do you really need the `$ns` variable? Can you just return results directly instead of assigning to $ns, and return null as a default at the end?

Reply:
Yes, and I simplified it further since it makes more sense to use as default whatever is parent node's default namespace instead of document's first namespace.